### ISSUE TITLE
Add Average Callback

### ISSUE THREAD

**Issue by Alexis-IMBERT** [Lightning-AI/pytorch-lightning#20617]
### Description & Motivation

Create a callback to allow creating an average (smoothing) of a logged metrics and log it. 
Why directly in lightning ? This feature could be useful for Early Stopping, Now in some training the loss (for example) is very unstable and you could get an exceptional minimum, but because it's very unstable, it's not the best epochs to stop.

Get an log average feature could smooth the loss (for example again) and allow better EarlyStopping, so better training

### Pitch

Create Callback class that get in parameter : the metrics to smooth, the name of the smoothed metric and the callable function to use to smooth the metric, the window, (how many previous values to consider). 
The callable function could be mean by default and window set to 5 by default.
